### PR TITLE
Flexible nav

### PR DESCRIPTION
**Pull request by mvandenbeuken**
:elephant: 

Allow an app using ama_layout to use its own nav yml file so that we can soft launch an app without changing the ama_layout gem.

### CODE REVIEW

**Comment by jbabe** [comment]
Do we want to `.gitignore` the `spec/internal/db/combustion_test.sqlite` file?

**Comment by mvandenbeuken** [comment]
@jbabe yeah. it just gets regenerated whenever rake is run. I think you and I ran into it being flagged as changed when we were testing out the html5 compliance fixes

**Comment by jbabe** [comment]
Use like this?

`Navigation.new nav_file_path: 'path/to/registries/file/in/registries/app/nav.yml'` ?

**Comment by mvandenbeuken** [comment]
@jbabe you've got it. I can update the README.md.

**Comment by jbabe** [comment]
yeah - i closed https://github.com/amaabca/ama_layout/tree/fix-html5-validation-errors until I re-open into the zurb PR in the correct repo

Maybe we just make the change here?

**Comment by jbabe** [comment]
i'm hoping to get back to the https://github.com/amaabca/ama_layout/tree/fix-html5-validation-errors before launch

**Comment by mvandenbeuken** [comment]
@jbabe for sure. This covers that sqlite database change:

+/spec/internal/db/*.sqlite in .gitignore. I git rm'd and added the gitignore line

**Comment by mvandenbeuken** [comment]
@jbabe. Updated the README.md for usage details

I'd love to get that HTML 5 validation change in before launch. I can help with that.

File: lib/ama_layout/navigation.rb [hunk 1]
```diff
@@ -26,8 +31,13 @@ def items
...
   private
     def navigation_items
-      file = File.join Gem.loaded_specs["ama_layout"].full_gem_path, "lib", "ama_layout", "navigation.yml"
-      YAML.load ERB.new(File.read file).result
+      YAML.load ERB.new(File.read nav_file_path).result
+    end
+
+    def defaults
+      {
+        nav_file_path: File.join(Gem.loaded_specs["ama_layout"].full_gem_path, "lib", "ama_layout", "navigation.yml")
```
Comment (jessedoyle) [inline lib/ama_layout/navigation.rb]: 👍 - Cool!

**Comment by jessedoyle** [comment]
🐘 🐘 🐘

**Pull request merged**